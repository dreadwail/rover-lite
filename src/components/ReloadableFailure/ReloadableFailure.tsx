import React, { FC } from 'react';

const reload = () => {
  const ignoreCache = true;
  window.location.reload(ignoreCache);
};

const ReloadableFailure: FC = () => (
  <div className="text-center">
    <p className="text-danger">An error occurred. Please try again in a moment.</p>
    <button type="button" className="btn btn-link" onClick={reload}>
      Click to reload.
    </button>
  </div>
);

export default ReloadableFailure;
