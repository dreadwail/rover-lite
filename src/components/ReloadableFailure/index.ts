export * from './ReloadableFailure';
export { default } from './ReloadableFailure';
