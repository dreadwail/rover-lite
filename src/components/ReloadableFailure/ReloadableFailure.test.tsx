import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';

import ReloadableFailure from './ReloadableFailure';

describe('ReloadableFailure', () => {
  let rendered: ShallowWrapper;

  beforeEach(() => {
    rendered = shallow(<ReloadableFailure />);
  });

  it('renders correctly', () => {
    expect(rendered).toMatchSnapshot();
  });

  describe('when the button is clicked', () => {
    beforeEach(() => {
      spyOn(window.location, 'reload');
      rendered.find('button').simulate('click');
    });

    it('reloads the page', () => {
      expect(window.location.reload).toHaveBeenCalled();
    });
  });
});
