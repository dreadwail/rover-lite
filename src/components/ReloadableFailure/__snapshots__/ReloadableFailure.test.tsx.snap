// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReloadableFailure renders correctly 1`] = `
<div
  className="text-center"
>
  <p
    className="text-danger"
  >
    An error occurred. Please try again in a moment.
  </p>
  <button
    className="btn btn-link"
    onClick={[Function]}
    type="button"
  >
    Click to reload.
  </button>
</div>
`;
